netkan/netkan/cli/utilities.py

```diff
@@ -0,0 +1,165 @@
+from ..auto_freezer import AutoFreezer
+
+
+@click.command()
+@click.option(
+    '--days-limit', default=1000,
+    help='Number of days to wait before freezing a mod as idle',
+)
+@common_options
+@pass_state
+def auto_freezer(common, days_limit):
+    afr = AutoFreezer(
+        common.netkan_remote,
+        common.github_pr,
+    )
+    afr.freeze_idle_mods(days_limit)
+    afr.mark_frozen_mods()
+
+
+@click.command()
+@common_options
+@pass_state
+def download_counter(common):
+    logging.info('Starting Download Count Calculation...')
+    DownloadCounter(
+        common.netkan_remote.working_dir,
+        common.ckanmeta_remote,
+        common.token
+    ).update_counts()
+    logging.info('Download Counter completed!')
+
+
+@click.command()
+@click.option(
+    '--status-bucket', envvar='STATUS_BUCKET', required=True,
+    help='Bucket to Dump status.json',
+)
+@click.option(
+    '--status-key', envvar='STATUS_KEY', default='status/netkan.json',
+    help='Overwrite bucket key, defaults to `status/netkan.json`',
+)
+@click.option(
+    '--interval', envvar='STATUS_INTERVAL', default=300,
+    help='Dump status to S3 every `interval` seconds',
+)
+def export_status_s3(status_bucket, status_key, interval):
+    frequency = 'every {} seconds'.format(
+        interval) if interval else 'once'
+    logging.info('Exporting to s3://%s/%s %s',
+                 status_bucket, status_key, frequency)
+    while True:
+        ModStatus.export_to_s3(status_bucket, status_key, interval)
+        if not interval:
+            break
+        time.sleep(interval)
+    logging.info('Done.')
+
+
+@click.command()
+def dump_status():
+    click.echo(json.dumps(ModStatus.export_all_mods()))
+
+
+@click.command()
+@click.argument('filename')
+def restore_status(filename):
+    click.echo(
+        'To keep within free tier rate limits, this could take some time'
+    )
+    ModStatus.restore_status(filename)
+    click.echo('Done!')
+
+
+@click.command()
+@common_options
+@pass_state
+def recover_status_timestamps(common):
+    ModStatus.recover_timestamps(common.ckanmeta_remote)
+
+
+@click.command()
+@click.option(
+    '--cluster', help='ECS Cluster running the service',
+)
+@click.option(
+    '--service-name', help='Name of ECS Service to restart',
+)
+def redeploy_service(cluster, service_name):
+    click.secho(
+        'Forcing redeployment of {}:{}'.format(cluster, service_name),
+        fg='green'
+    )
+    client = boto3.client('ecs')
+    services = client.list_services(maxResults=100,
+                                    cluster=cluster)['serviceArns']
+    try:
+        service = list(filter(lambda i: service_name in i, services))[0]
+    except IndexError:
+        available = '\n    - '.join(
+            [f.split('/')[1].split('-')[1] for f in services]
+        )
+        raise click.UsageError(
+            "Service '{}' not found. Available services:\n    - {}".format(
+                service, available)
```

Comment:
This hasn't changed in your PR, but I just spotted it:
Shouldn't this be `sevice_name` instead of `service`? `service` will be unassigned when an exception is thrown, and I think we want to output the service name that was asked for.

Reply:
```
Uncaught exception:
Traceback (most recent call last):
  File "/home/netkan/.local/lib/python3.7/site-packages/netkan/cli.py", line 203, in redeploy_service
    service = list(filter(lambda i: service_name in i, services))[0]
IndexError: list index out of range

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/netkan/.local/bin/netkan", line 8, in <module>
    sys.exit(netkan())
  File "/home/netkan/.local/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/netkan/.local/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/netkan/.local/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/netkan/.local/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/netkan/.local/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/netkan/.local/lib/python3.7/site-packages/netkan/cli.py", line 210, in redeploy_service
    service, available)
UnboundLocalError: local variable 'service' referenced before assignment
```

I think this confirms my theory :)

Reply:
Yes.. You're quite right! Good spot :slightly_smiling_face: 

```
$ netkan redeploy-service --cluster NetKANCluster --service-name WhatIsYourFavouriteColour
[2020-01-25 01:51:41,997] [INFO    ] Logging started for 'redeploy-service' at log level 20
Forcing redeployment of NetKANCluster:WhatIsYourFavouriteColour
[2020-01-25 01:51:43,098] [ERROR   ] Uncaught exception:
Traceback (most recent call last):
  File "/home/netkan/.local/lib/python3.7/site-packages/netkan/cli.py", line 203, in redeploy_service
    service = list(filter(lambda i: service_name in i, services))[0]
IndexError: list index out of range

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/netkan/.local/bin/netkan", line 8, in <module>
    sys.exit(netkan())
  File "/home/netkan/.local/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/netkan/.local/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/netkan/.local/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/netkan/.local/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/netkan/.local/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/netkan/.local/lib/python3.7/site-packages/netkan/cli.py", line 210, in redeploy_service
    service, available)
UnboundLocalError: local variable 'service' referenced before assignment
Traceback (most recent call last):
  File "/home/netkan/.local/lib/python3.7/site-packages/netkan/cli.py", line 203, in redeploy_service
    service = list(filter(lambda i: service_name in i, services))[0]
IndexError: list index out of range

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/netkan/.local/bin/netkan", line 8, in <module>
    sys.exit(netkan())
  File "/home/netkan/.local/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/netkan/.local/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/netkan/.local/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/netkan/.local/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/netkan/.local/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/netkan/.local/lib/python3.7/site-packages/netkan/cli.py", line 210, in redeploy_service
    service, available)
UnboundLocalError: local variable 'service' referenced before assignment
```

Reply:
This change got lost in the rebase.

Reply:
Ahh crumbs! That rebase wasn't very straight forward, good catch.. again!